runtime-light/stdlib/file/file-system-functions.h

```diff
@@ -180,9 +188,38 @@ inline resource f$stream_socket_client(const string& address, std::optional<std:
 }
 
 inline Optional<string> f$file_get_contents(const string& stream) noexcept {
-  if (auto sync_resource{from_mixed<class_instance<kphp::fs::sync_resource>>(f$fopen(stream, {}), {})}; !sync_resource.is_null()) {
+  if (auto sync_resource{from_mixed<class_instance<kphp::fs::sync_resource>>(
```

Comment:
As we are sure this works for files only, it might be worth it to open a file directly. Thus, we will eliminate a dynamic cast and `string` creation

Reply:
This function works for both `"php://input"` and files.